client/src/model/sensor.ts

```diff
@@ -4,3 +4,9 @@ export interface Sensor {
   type: 'A' | 'B' | 'C'
   extra?: string
 }
+
+export interface SensorCreateParams {
+  display_name: string
```

Comment:
I strongly suggest do not mix naming style of snake_case and CamelCase in the same project. Bad smell.

Reply:
Thanks for @aMikeChen , mixing `snake_case` and `camelCase` is not good. But there is a convention in a frontend project that every variable naming with `snake_case` means it's defined from backend

Because we did teach this convention in the lesson, it's up to @aChloeShen to define the variable naming style

Reply:
@aYoyoPan Just a little advise. 
In this case, if we choose OOP style, I will define a **Class** for converting API response data to object used in components, first I will define each model class in different files:
```js
// client/src/model/SensorCreateParams.ts
export class SensorCreateParams {
  readonly displayName: string
  ...

  constructor(apiResponse :: object) {
    this.displayName = apiResponse["display_name"]
  }
}

// import { SensorCreateParams } from '../../model/SensorCreateParams'
// const sensorCreateParams = new SensorCreateParams(res.body)
```
Then we can add some validation or calculation when converting API response.
Or you can make a general converter as a middleware if it's just 1 to 1 mapping.
For data object, I will use `data["data_key"]` but not `data.data_key`.

Reply:
@aMikeChen @aYoyoPan @aChloeShen 

IIRC, there was some discussion about converting the response from backend (`snake_case`) to frontend (`camelCase`).
There are lots of packages available to achieve this goal (`lodash` is not an option to me as it's heavy loading).
Or, we can even make this from scratch.